vendor/github.com/jesseduffield/gocui/view.go

```diff
@@ -996,8 +1002,36 @@ func (v *View) updateSearchPositions() {
 			return result
 		}
 
-		for y, line := range v.lines {
-			v.searcher.searchPositions = append(v.searcher.searchPositions, searchPositionsForLine(line, y)...)
+		if v.searcher.modelSearchFunc != nil {
+			modelSearchResults := v.searcher.modelSearchFunc(normalizedSearchStr, caseSensitive)
```

Comment:
I'm realizing that calling the model search func every time we call updateSearchPositions is not ideal, because it's unnecessarily slow in cases where the model hasn't changed. For example, when searching in a list of 1 million commits (e.g. the linux kernel repo), moving the selection up/down rerenders the view in order to render the highlight of the commit graph; this calls updateSearchPositions, so we unnecessarily search the model again every time the selection changes. This makes interacting with the view extremely sluggish.

Maybe a better design would be to not set a search func, but pass an optional array of model search positions in to the call to Search. This would make it necessary to call Search again when the model changes, but since we do this already for filterable views, this should be easy to add (see SearchHelper.ReApplyFilter).

@jesseduffield Any opinions?

Reply:
I went ahead and made that change, I think it's the better approach. I did this in a few fixups (a3031f6fec67 and 3f12ab43782c, and 42901e932321 for an unrelated performance improvement).

Since I assume you haven't started to review this yet, I'll squash them right away, should be easier to review then.

Reply:
That approach makes sense to me